### DISCUSSION TITLE
issue: OnChange event does not fire for HTML select within a React Hook Form

### DISCUSSION THREAD

**Discussion by floracode1** [react-hook-form/react-hook-form#8672] [General] [answered]
### Version Number

7.31.2

### Codesandbox/Expo snack

[none](https://snack.expo.dev/@flora.liao/select-react-hook-form)

### Steps to reproduce

1. Create a select drop down within a react hook form. E.g.
                          <select
                                id="province"
                                className={`form-control ${errors.province
                                    ? "has-error"
                                    : ""
                                    }`}
                                name="province

                                onChange={(e) => handleSelectChange(e)}
                                {...register("province", registeredOptions.province)}
                            >
                                <option value="">Please Select</option>
                                <option value="ON">Ontario</option>
                                <option value="QB">Quebec</option>
                            </select>
2. Create an onChange event for this select drop down. 
3. Event does not fire. Change event to onClick and it works. Move select drop down outside of the react hook form and it works as well. 

### Expected behaviour

onChange event should fire for dropdown value change

### What browsers are you seeing the problem on?

Chrome

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by Moshyfawn** [comment]
Temporary closing as "need more info"

Please, take your time to provide **a minimal** reproduction codesandbox

**Comment by floracode1** [comment]
Please see the example below: 
https://snack.expo.dev/@flora.liao/select-react-hook-form

Change the onChange event on line 32 to onClick and the alert is triggered, but not for onChange

**Comment by floracode1** [comment]
> Please see the example below: https://snack.expo.dev/@flora.liao/select-react-hook-form
> 
> Change the onChange event on line 32 to onClick and the alert is triggered, but not for onChange

**Comment by bluebill1049** [comment, marked as answer]
```tsx
onChange={(e) => alert("clicked")}
{...register("province")}
```

it's getting overwritten: https://react-hook-form.com/api/useform/register

![Screen Shot 2022-07-13 at 10 05 57 am](https://user-images.githubusercontent.com/10513364/178619135-9c29f3b9-3bc2-4571-bfe2-1033f2c7179b.png)

**Comment by floracode1** [comment]
cool, thank you!
